fix(view): Show the correct publish date for versions selected by range

andersk opened the pull request:
Before, `npm view npm@^6` would incorrectly report “published over a year from now” for every entry. Now it reports the correct dates.

wraithgar commented on test/lib/view.js:
```diff
@@ -332,6 +332,13 @@ t.test('should log package info', t => {
 
+  t.test('package with semver range', t => {
+    view.exec(['blue@^1.0.0'], () => {
+      t.matchSnapshot(logs)
```
We should set blue's time object up so that this snapshot tests what we want here.

So before mocks we can set up 
```javascript
// 25 hours ago
const yesterday = new Date(new Date() - 1000 * 60 * 60 * 25)
```

and blue's time object can be

```javascript
      time: {
        '1.0.0': yesterday,
      },
```

And then our snapshots for blue will all say `published yesterday`

andersk replied:
Adjusted.

wraithgar commented:
Really good catch here, thank you. Just one note on making sure we're testing this new behavior, but other than that this is obviously the correct solution.

andersk committed:
fix(view): Show the correct publish date for versions selected by range

Before, `npm view npm@^6` would incorrectly report “published over a
year from now” for every entry.  Now it reports the correct dates.

PR-URL: https://github.com/npm/cli/pull/3739
Credit: @andersk
Close: #3739
Reviewed-by: @wraithgar

wraithgar reviewed (approved):
Thank you for submitting this
